### ISSUE TITLE
Allow input labels to be logits for knowledge distilation

### ISSUE THREAD

**Issue by felipemello1** [pytorch/torchtune#2007]
Users may want to run offline inference using a larger model, e.g. llama 405B, and train a smaller model using it. We currently have a knowledge distilation recipe that does it online, i.e. for every step, the teacher and the student run on the same input.

However, enabling offline inference can be an optimal way to do it, if the teacher is frozen.

If this makes sense, we would have to update our recipes/datasets to properly support it.

cc: @lindawangg
